flink-runtime-web/web-dashboard/src/app/share/customize/flame-graph/flame-graph.component.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, ChangeDetectionStrategy, ElementRef, Input, ViewChild } from '@angular/core';
+import { JobFlameGraphNodeInterface } from 'interfaces';
+import { select } from 'd3-selection';
+import { format } from 'd3-format';
+import { flamegraph, offCpuColorMapper } from 'd3-flame-graph';
+
+import * as _d3 from 'd3';
```

Comment:
there is an angular native flamegraph here https://github.com/mgechev/ngx-flamegraph, not sure if it can meet your requirements, but you can have a try.

Reply:
I'd suggest to do a follow-up PR for that.